imap_processing/cdf/utils.py

```diff
@@ -12,15 +12,15 @@
 logger = logging.getLogger(__name__)
 
 
-def calc_start_time(shcoarse_time: int):
+def calc_start_time(shcoarse_time: float) -> np.datetime64:
     """Calculate the datetime64 from the CCSDS secondary header information.
 
     Since all instrument has SHCOARSE or MET seconds, we need convert it to
     UTC. Took this from IDEX code.
 
     Parameters
     ----------
-    shcoarse_time: int
+    shcoarse_time: float
```

Comment:
This is used not for SHCOARSE time, but for a more precise time from the MAG instrument which includes fractional seconds. But, this is a good point, that changing the type here might affect other instruments. I'm happy to make a different method if you'd prefer to keep the SHCOARSE separate from other spacecraft time conversions.

Reply:
Alternatively, I could make it clear in this function that it isn't just for SHCOARSE but for other types of time conversions? I assumed that we will eventually replace this with SPICE time conversions, so I could borrow it in a temporary way for MAG processing too.

Reply:
Since all tests passed, I think it's ok to leave it as it is. We are going to need to refactor this function or implement another function do below code since many of us are doing this similar thing. We can handle this in future PR.
```
epoch_converted_time = [calc_start_time(time) for time in epoch_time]
```